src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -1725,7 +1746,22 @@ public void updateTriggerConfigURL() {
                         + "a specific configured server");
             } else {
                 List<GerritProject> fetchedProjects = GerritDynamicUrlProcessor.fetch(triggerConfigURL, serverName);
-                dynamicGerritProjects = fetchedProjects;
+                // Since other threads may be accessing the dynamic project list, we need to make sure
+                // that we update its contents in a thread-safe way.
+                synchronized (DYNAMIC_GERRIT_PROJECTS_LOCK) {
+                    if (dynamicGerritProjects == null) {
+                        logger.debug("Project number changed from null to "
```

Comment:
String concatenation for a debug log is bad. Use `{}` formating instead.

Reply:
I did, but I got a compile-time error having to do with not being able to use a string as `Marker` or _vice versa_ (I forget which); I suspect that it has to do with the `debug(Marker marker, String format, Object arg1, Object arg2)` signature.

Reply:
if you need more than two args you can do `new Object[]{arg1, arg2, arg3}`